pypowerwall/cloud/pypowerwall_cloud.py

```diff
@@ -0,0 +1,997 @@
+            if island_status == "on_grid":
+                alert = "SystemConnectedToGrid"
+            elif island_status == "off_grid_intentional":
+                alert = "ScheduledIslandContactorOpen"
+            elif island_status == "off_grid":
+                alert = "UnscheduledIslandContactorOpen"
+            else:
+                alert = ""
+                if lookup(power, ("response", "grid_status")) == "Active":
+                    alert = "SystemConnectedToGrid"
+            data = {
+                f'STSTSM--{part_number}--{serial_number}': {
+                    'partNumber': part_number,
+                    'serialNumber': serial_number,
+                    'manufacturer': 'Simulated',
+                    'firmwareVersion': version,
+                    'lastCommunicationTime': int(time.time()),
+                    'teslaEnergyEcuAttributes': {
+                        'ecuType': 207
+                    },
+                    'STSTSM-Location': 'Simulated',
+                    'alerts': [
+                        alert
+                    ]
+                }
+            }
+        return data
+
+    def get_api_meters_aggregates(self) -> Optional[Union[dict, list, str, bytes]]:
+        config = self.get_site_config()
+        power = self.get_site_power()
+        if config is None or power is None:
+            data = None
+        else:
+            timestamp = lookup(power, ("response", "timestamp"))
+            solar_power = lookup(power, ("response", "solar_power"))
+            battery_power = lookup(power, ("response", "battery_power"))
+            load_power = lookup(power, ("response", "load_power"))
+            grid_power = lookup(power, ("response", "grid_power"))
+            battery_count = lookup(config, ("response", "battery_count"))
+            inverters = lookup(config, ("response", "components", "inverters"))
+            if inverters is not None:
+                solar_inverters = len(inverters)
+            elif lookup(config, ("response", "components", "solar")):
+                solar_inverters = 1
+            else:
+                solar_inverters = 0
+            data = API_METERS_AGGREGATES_STUB
+            data['site'].update({
+                "last_communication_time": timestamp,
+                "instant_power": grid_power,
+            })
+            data['battery'].update({
+                "last_communication_time": timestamp,
+                "instant_power": battery_power,
+                "num_meters_aggregated": battery_count,
+            })
+            data['load'].update({
+                "last_communication_time": timestamp,
+                "instant_power": load_power,
+
+            })
+            data['solar'].update({
+                "last_communication_time": timestamp,
+                "instant_power": solar_power,
+                "num_meters_aggregated": solar_inverters,
+            })
```

Comment:
Moved large dictionaries to a separate file with stubs and just update the fields we have information for.